controller/src/beerocks/master/db/db.cpp

```diff
@@ -2403,7 +2403,8 @@ bool db::add_client_to_persistent_db(const sMacAddr &mac, const ValuesMap &param
         return false;
     }
 
-    if (m_persistent_db_clients_count >= config.clients_persistent_db_max_size) {
+    while ((m_persistent_db_clients_count >= config.clients_persistent_db_max_size) &&
+           (m_persistent_db_clients_count > 0)) {
```

Comment:
If `m_persistent_db_clients_count >= config.clients_persistent_db_max_size` is `true` then, `m_persistent_db_clients_count > 0` must be `true` too because `config.clients_persistent_db_max_size` will always be greater than 0.
So this makes the  `m_persistent_db_clients_count > 0` redundant. Correct me if I'm wrong.

Reply:
the condition should be `config.clients_persistent_db_max_size > 0` - thanks for pointing that out.

Reply:
in the next commit this condition has actually changed to
```suggestion
    while ((m_persistent_db_clients_count >= config.clients_persistent_db_max_size) &&
           ((m_persistent_db_clients_count > 0) || (config.clients_persistent_db_max_size == 0))) {
```
but looking at it now I should probably take care of this as separate if condition at the beginning of the function